include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -148,56 +148,56 @@ __sub_group_scan_partial(const _SubGroup& __sub_group, _ValueType& __value, _Bin
 }
 
 template <std::uint8_t __sub_group_size, bool __is_inclusive, bool __init_present, bool __capture_output,
-          std::uint32_t __max_inputs_per_item, typename _SubGroup, typename _UnaryOp, typename _BinaryOp,
-          typename _LazyValueType, typename _InRng, typename _OutRng>
+          std::uint32_t __max_inputs_per_item, typename _SubGroup, typename _GenInput, typename _ScanPred,
+          typename _BinaryOp, typename _WriteOp, typename _LazyValueType, typename _InRng, typename _OutRng>
 void
-__scan_through_elements_helper(const _SubGroup& __sub_group, _UnaryOp __unary_op, _BinaryOp __binary_op,
-                               _LazyValueType& __sub_group_carry, _InRng __in_rng, _OutRng __out_rng,
-                               std::size_t __start_idx, std::size_t __n, std::uint32_t __iters_per_item,
-                               std::size_t __subgroup_start_idx, std::uint32_t __sub_group_id,
-                               std::uint32_t __active_subgroups)
+__scan_through_elements_helper(const _SubGroup& __sub_group, _GenInput __gen_input, _ScanPred __scan_pred,
```

Comment:
When I see pred or predicate I usually think of a function that returns true or false which confused me a bit when I saw this at first. Naming this something like `__scan_op` also sounds confusing though. How about something like `__scan_input_transform`?

Reply:
Although I do see why it is named `__scan_pred` since the unary predicate would be applied here for some scan-based algorithms, so it is difficult to think of a good name for this.

Reply:
Yes, I agree.  Your suggestion looks good.  

Another option is `_ScanUnaryOp` __scan_unary_op`, but I think I like your option better.

Reply:
applied.

Reply:
applied.